test/new-e2e/tests/agent-platform/common/test_client.go

```diff
@@ -281,3 +282,93 @@ func ReadJournalCtl(t *testing.T, client *TestClient, grepPattern string) string
 	}
 	return journalCtlOutput
 }
+
+// DockerTestClient is a helper to run commands on a docker container for tests
+type DockerTestClient struct {
+	host          *components.RemoteHost
+	containerName string
+}
+
+// NewDockerTestClient creates a client to help write tests that run on a docker container
+func NewDockerTestClient(host *components.RemoteHost, containerName string) DockerTestClient {
+	return DockerTestClient{
+		host:          host,
+		containerName: containerName,
+	}
+}
+
+// RunContainer starts the docker container in the background based on the given image reference
+func (c *DockerTestClient) RunContainer(image string) error {
+	// We run an infinite no-op to keep it alive
+	_, err := c.host.Execute(
+		fmt.Sprintf("docker run -d -e DD_HOSTNAME=docker-test --name '%s' '%s' tail -f /dev/null", c.containerName, image),
+	)
+	return err
+}
+
+// Cleanup force-removes the docker container associated to the client
+func (c *DockerTestClient) Cleanup() error {
+	_, err := c.host.Execute(fmt.Sprintf("docker rm -f '%s'", c.containerName))
+	return err
+}
+
+// Execute runs commands on a Docker remote host
+func (c *DockerTestClient) Execute(command string) (output string, err error) {
+	return c.host.Execute(
+		// Run command on container via docker exec,
+		// wrap in a shell call to achieve similar behavior to remote hosts
+		fmt.Sprintf("docker exec %s sh -c '%s'", c.containerName, command),
+	)
+}
+
+// ExecuteWithRetry execute the command with retry
+func (c *DockerTestClient) ExecuteWithRetry(command string) (output string, err error) {
+	for try := 0; try < 5; try++ {
+		output, err = c.Execute(command)
+		if err == nil {
+			break
+		}
+		time.Sleep(time.Duration(math.Pow(2, float64(try))) * time.Second)
+	}
+
+	return output, err
+}
+
+// InstallWithInstallScript tries to run the install script on the docker host.
+// It makes the test fail if the installation fails
+func (c *DockerTestClient) InstallWithInstallScript(t *testing.T) {
```

Comment:
*nit*: Just a thought that this method seems to be different enough from the other methods for the client as it actually holds the tests. Would it make sense to split into a different struct that receives a "Client" that can `Execute` commands themselves?

Reply:
Maybe not a struct but a function on the `install` package like in https://github.com/DataDog/datadog-agent/blob/ca5dd2d2b039a18e8dfddf2169000ba1fe4d4282/test/new-e2e/tests/agent-platform/install-script/install_script_test.go#L146

That sounds fair, I guess I was trying too hard to not change as many files (as with the other PR).

Reply:
Addressed now, but making the `install.Unix` function more generic by accepting an interface; this change is small and doesn't impact any callers so it's still pretty small in scope.